BitmarkRegistry/Services/FileCourierService.swift

```diff
@@ -0,0 +1,168 @@
+      ]
+
+      let assetData = try Data(contentsOf: encryptedFileURL)
+      Alamofire.upload(multipartFormData: { (multipartFormData) in
+        for (key, value) in parameters {
+          multipartFormData.append("\(value)".data(using: .utf8)!, withName: key)
+        }
+
+        let mineType = getMineType(of: encryptedFileURL)
+        multipartFormData.append(assetData, withName: "file", fileName: assetFilename, mimeType: mineType)
+      }, usingThreshold: UInt64(), to: uploadURL, method: .post, headers: headers) { (result) in
+        switch result{
+        case .success(let upload, _, _):
+          upload.responseJSON { response in
+            if let error = response.error {
+              ErrorReporting.report(error: error)
+            }
+          }
+        case .failure(let error):
+          ErrorReporting.report(error: error)
+        }
+      }
+    } catch {
+      ErrorReporting.report(error: error)
+    }
+  }
+
+  static func getDownloadableAssets(receiver: Account, completion: @escaping (Array<String>?, Error?) -> Void) {
+    guard let jwt = KeychainStore.getJwtFromKeychain() else { return }
+
+    let url = URL(string: Global.ServerURL.fileCourier + "/v2/files?receiver=" + receiver.getAccountNumber())!
+    var request = URLRequest(url: url)
+    request.allHTTPHeaderFields = [
+      "Accept" : "application/json",
+      "Content-Type" : "application/json",
+      "Authorization" : "Bearer " + jwt
+    ]
+
+    URLSession.shared.dataTask(with: request) { (data, response, error) in
+      if let error = error {
+        completion(nil, error); return
+      }
+
+      if let data = data {
+        do {
+          let jsonObject = try JSONSerialization.jsonObject(with: data) as! [String : Array<String>]
+          let downloadableFileInfos = jsonObject["file_ids"]
+
+          // validate file_ids response
+          downloadableFileInfos?.forEach({ (info) in
+            if !isValidDownloadableInfo(info) {
+              completion(nil, nil)
+            }
+          })
+
+          completion(downloadableFileInfos, nil)
+        } catch {
+          completion(nil, error)
+        }
+      }
+    }.resume()
+  }
+
+  typealias responseData = (sessionData: SessionData, filename: String, encryptedFileData: Data)
+  static func downloadFileFromCourierServer(
+    assetId: String, receiver: Account,
+    senderAccountNumber: String, senderPublicKey: Data,
+    completion: @escaping (responseData?, Error?) -> Void) {
+
+    guard let jwt = KeychainStore.getJwtFromKeychain() else { return }
+
+    let downloadURL = URL(string: Global.ServerURL.fileCourier + "/v2/files/" + assetId + "/" + senderAccountNumber + "?receiver=" + receiver.getAccountNumber())!
+    var downloadRequest = URLRequest(url: downloadURL)
+    downloadRequest.allHTTPHeaderFields = [
+      "Authorization": "Bearer " + jwt
+    ]
+
+    URLSession.shared.downloadTask(with: downloadRequest) { (tempLocalURL, response, error) in
```

Comment:
Later on, we can refactor to make using `Alamofire` only for the requests.

Reply:
This for sharing my intent:
I did want to make the discovery journey to use URLSession instead of Alamofire to understand how Request look like. 🍡 
But then the upload file request make too much code and especially it's not working (maybe with new request formData structure)
That's why I move to use Alamofire😸